### PR TITLE
Replace legacy build pipeline with PowerShell SDP build pipeline for SharePoint docs

### PR DESCRIPTION
**Pull request by jeffreyxrh**
#6329

### CODE REVIEW

File: sharepoint/docfx.json [hunk 1]
```diff
@@ -53,25 +66,25 @@
...
+      "ms.devlang": "powershell",
       "feedback_system": "GitHub",
       "feedback_github_repo": "MicrosoftDocs/office-docs-powershell",
       "feedback_product_url": "https://github.com/MicrosoftDocs/office-docs-powershell/issues"
     },
     "fileMetadata": {
-      "ms.service":{
-        "sharepoint-ps/sharepoint-online/*.md":"sharepoint-powershell"
+      "ms.service": {
+        "sharepoint-ps/sharepoint-online/*.md": "sharepoint-powershell"
```
Comment (qinezh) [inline sharepoint/docfx.json:81]: => `yml`

File: sharepoint/docfx.json [hunk 2]
```diff
@@ -2,7 +2,10 @@
   "build": {
     "content": [
       {
-        "files": ["**/*.md","**/**.yml"],
+        "files": [
```
Comment (qinezh) [inline sharepoint/docfx.json:5]: Let's just keep the format for better PR review experience.

**Comment by qinezh** [comment]
> we were informed that Json files are protected and managed by admins only, please follow up with a direct mail to the vendor about what to do next, contacting admins at marvel docs.

@rloutlaw can you please help on this?

**Comment by rloutlaw** [comment]
@dariomws @get-itips I'm a admin for the MicrosoftDocs GitHub organization-these changes to the publishing JSON files are a part of regular feature work and updates by the reference engineering team.

There is no customer experience change as part of this work, and your docs that are live today will not be affected on the live site. They will be generated through a different toolchain with the same outcome.

I will merge this PR later in the business day today to unblock @qinezh 's work.

**Comment by rloutlaw** [comment]
@dariomws @get-itips - When we merge this PR, we're likely to introduce 21 warnings due to broken links (all relative path links to .md files, which aren't published directly now but are instead processed to yml and then published from that). There is an XREF syntax that the links need to be updated-see the change made in one of the markdowns in this PR.

**Comment by dstrome** [comment]
@rloutlaw - I'm sending you an email regarding this PR. Please don't merge this or the other SDP PR yet.

**Pull request merged**